pkg/sql/colexec/mergeorder/order.go

```diff
@@ -158,8 +158,14 @@ func mergeSort(proc *process.Process, bat2 *batch.Batch,
 }
 
 func (ctr *container) mergeSort2(bat2 *batch.Batch, proc *process.Process) error {
+	var err error
 	if ctr.bat == nil {
 		ctr.bat = bat2
+		ctr.bat, err = bat2.Dup(proc.Mp())
```

Comment:
ctr.bat = bat2
ctr.bat.Expand() will be suitable.

but I'm not sure if we have the Expand method of batch.